packages/devtools_app/lib/src/error_badge_manager.dart

```diff
@@ -43,16 +44,52 @@ class ErrorBadgeManager extends DisposableController
     if (e.extensionKind == 'Flutter.Error') {
       incrementBadgeCount(LoggingScreen.id);
 
-      final json = e.extensionData.data;
-      final objectId = json['objectId'] as String;
-      // TODO(kenz): verify that the object id is referring to an element or
-      // widget.
-      if (objectId?.contains('inspector-') ?? false) {
-        incrementBadgeCount(InspectorScreen.id);
+      final inspectableError = _extractInspectableError(e);
+      if (inspectableError != null) {
+        appendInspectorErroredWidget(inspectableError);
       }
     }
   }
 
+  InspectableWidgetError _extractInspectableError(Event error) {
```

Comment:
Use the existing RemoteDiagnosticsNode class to parse these objects instead of writing a custom json parser here.

Reply:
Done in a0f58d79, although I'm passing `null` as the `inspectorService` (it wasn't clear how I can get one here, and it didn't seem to be required) and I'm not sure if that's reasonable?

Reply:
^^ this is still outstanding. It seems to work, but I'm not sure it's valid to pass `null` here (and if not, what the correct way to get a reference to an `inspectorService` here would be.

Reply:
@jacob314? ☝️

Reply:
Passing a null inspectorService is ok. Add a TODO, I'm about to add inspectorService to the serviceManager so it is always available if you have a Flutter app.

Reply:
Added in 257f2cbe